src/rust/generator-dispatcher/tests/test_utils/context.rs

```diff
@@ -28,27 +28,16 @@ impl AsyncTestContext for GeneratorDispatcherTestContext {
     async fn setup() -> Self {
         let _guard = setup_tracing(SERVICE_NAME).expect("setup_tracing");
 
-        let endpoint = std::env::var("PIPELINE_INGRESS_CLIENT_ADDRESS")
-            .expect("missing environment variable PIPELINE_INGRESS_CLIENT_ADDRESS");
-
-        tracing::info!(
-            message = "waiting 10s for pipeline-ingress to report healthy",
-            endpoint = %endpoint,
-        );
-
-        HealthcheckClient::wait_until_healthy(
-            endpoint.clone(),
-            PipelineIngressClient::SERVICE_NAME,
-            Duration::from_millis(10000),
-            Duration::from_millis(500),
+        let client_config = PipelineIngressClientConfig::parse();
+        let pipeline_ingress_client = get_grpc_client_with_options(
+            client_config,
+            GetGrpcClientOptions {
+                perform_healthcheck: true,
+                ..Default::default()
+            },
         )
         .await
-        .expect("pipeline-ingress never reported healthy");
-
-        tracing::info!("connecting pipeline-ingress gRPC client");
-        let pipeline_ingress_client = PipelineIngressClient::connect(endpoint.clone())
-            .await
-            .expect("could not configure gRPC client");
+        .expect("pipeline_ingress_client");
```

Comment:
this message doesn't tell me what _happened_, whereas the previous one (in the context of the logged messages) tells me exactly what went wrong.. we waited for 10s but the service never reported healthy.